File: app/src/main/java/at/shockbytes/coins/network/PriceManager.java
package at.shockbytes.coins.network;

import at.shockbytes.coins.currency.CryptoCurrency;
import at.shockbytes.coins.currency.Currency;
import at.shockbytes.coins.network.model.PriceConversion;
import rx.Observable;

/**
 * @author Martin Macheiner
 *         Date: 14.06.2017.
 */

public interface PriceManager {

    Observable<PriceConversion> getSpotPrice(CryptoCurrency from, Currency to);

    Observable<PriceConversion> getBuyPrice(CryptoCurrency from, Currency to);

    Observable<PriceConversion> getSellPrice(CryptoCurrency from, Currency to);

    boolean supportsCurrencyConversion(CryptoCurrency currency);

}

File: app/src/main/java/at/shockbytes/coins/network/PriceManager.kt
package at.shockbytes.coins.network

import at.shockbytes.coins.currency.CryptoCurrency
import at.shockbytes.coins.currency.Currency
import at.shockbytes.coins.network.model.PriceConversion
import io.reactivex.Observable

/**
 * @author Martin Macheiner
 * Date: 14.06.2017.
 */

interface PriceManager {

    val name: String

    val icon: Int

    var isEnabled: Boolean

    fun getSpotPrice(from: CryptoCurrency, to: Currency): Observable<PriceConversion>

    fun getBuyPrice(from: CryptoCurrency, to: Currency): Observable<PriceConversion>

    fun getSellPrice(from: CryptoCurrency, to: Currency): Observable<PriceConversion>

    fun supportsCurrencyConversion(currency: CryptoCurrency): Boolean

}
